modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeGetImagePreviewURLMethodCheck.java

```diff
@@ -24,44 +24,48 @@
 /**
  * @author Tamyris Bernardo
  */
-public class UpgradeGetImagePreviewURLMethodCheck extends BaseFileCheck {
+public class UpgradeGetImagePreviewURLMethodCheck
+	extends BaseUpgradeMatcherReplacementCheck {
 
 	@Override
-	protected String doProcess(
-			String fileName, String absolutePath, String content)
-		throws Exception {
+	protected String afterFormat(
+		String fileName, String absolutePath, String content,
+		String newContent) {
 
-		if (!fileName.endsWith(".java") && !fileName.endsWith(".jsp")) {
-			return content;
+		if (fileName.endsWith(".java")) {
+			newContent = JavaSourceUtil.addImports(
+				newContent, "com.liferay.document.library.util.DLURLHelper");
```

Comment:
Do you think `BaseUpgradeMatcherReplacementCheck` could benefit from having a method for dealing with imports specifically? I was thinking we could dedicated methods for adding/removing/replacing imports.

This might overcomplicate the check. But let me know your thoughts on this idea.

Reply:
For example, it would look something like this:

```java
@Override
protected String doProcess(...) {
    ...

    changeImports();
}

// Checks would override this method if necessary
protected void changeImports() {
    deleteImports(...);
    addImports(...);
    replaceImports(...);
}


```

Reply:
This method should return the modified content by the other methods. And the inside methods should be abstracts to be overwritten when the child check needs to use?

Reply:
Fair enough. Just an idea.